cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityStatusStore.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.capability;
+
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import io.cdap.cdap.api.dataset.lib.CloseableIterator;
+import io.cdap.cdap.spi.data.StructuredRow;
+import io.cdap.cdap.spi.data.StructuredTable;
+import io.cdap.cdap.spi.data.table.field.Field;
+import io.cdap.cdap.spi.data.table.field.Fields;
+import io.cdap.cdap.spi.data.table.field.Range;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import io.cdap.cdap.store.StoreDefinition;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * CapabilityStatusStore which takes care of reading , writing capability status and provides additional helpful methods
+ */
+public class CapabilityStatusStore implements CapabilityReader, CapabilityWriter {
+
+  private static final Gson GSON = new Gson();
+  private final TransactionRunner transactionRunner;
+
+  @Inject
+  CapabilityStatusStore(TransactionRunner transactionRunner) {
+    this.transactionRunner = transactionRunner;
+  }
+
+  @Override
+  public void checkAllEnabled(Collection<String> capabilities) throws IOException, CapabilityNotAvailableException {
+    List<List<Field<?>>> multiKeys = new ArrayList<>();
+    capabilities.forEach(capability -> multiKeys
+      .add(Collections.singletonList(Fields.stringField(StoreDefinition.CapabilitiesStore.NAME_FIELD, capability))));
+    Map<String, String> capabilityMap = TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.CapabilitiesStore.CAPABILITIES);
+      return capabilityTable.multiRead(multiKeys).stream().
+        collect(Collectors.toMap(
+          structuredRow -> structuredRow.getString(StoreDefinition.CapabilitiesStore.NAME_FIELD),
+          structuredRow -> structuredRow.getString(StoreDefinition.CapabilitiesStore.STATUS_FIELD)
+        ));
+
+    }, IOException.class, CapabilityNotAvailableException.class);
```

Comment:
I don't think it will throw the `CapabilityNotAvailableException` from the transaction, right?

Reply:
It doesn't, removed.